src/commons/affinities.rs

```diff
@@ -0,0 +1,398 @@
+    serde(rename_all = "camelCase")
+)]
+pub struct StackableAffinity {
+    pub pod_affinity: Option<PodAffinity>,
+    pub pod_anti_affinity: Option<PodAntiAffinity>,
+    pub node_affinity: Option<NodeAffinity>,
+    pub node_selector: Option<StackableNodeSelector>,
+}
+
+impl StackableAffinityFragment {
+    #[deprecated(
+        since = "0.36.0",
+        note = "During https://github.com/stackabletech/issues/issues/323 we moved from the previous selector field on a rolegroup to a more generic affinity handling. \
+We still need to support the old selector field, which has some custom magic (see the code in this function). \
+So we need a way to transform the old into the mechanism which this function offers. \
+It will be removed once we stop supporting the old mechanism."
+    )]
+    pub fn add_legacy_selector(&mut self, label_selector: &LabelSelector) {
+        tracing::error!("Deprecated field `selector` was specified. Please use the new `affinity` field instead, as support for `selector` will be removed in the future. See https://docs.stackable.tech/home/stable/contributor/adr/ADR026-affinities.html for details");
+
+        let node_labels = label_selector.match_labels.clone();
+        let node_label_exprs = label_selector.match_expressions.clone();
+
+        let node_affinity = node_label_exprs.map(|node_label_exprs| NodeAffinity {
+            required_during_scheduling_ignored_during_execution: Some(NodeSelector {
+                node_selector_terms: vec![NodeSelectorTerm {
+                    match_expressions: Some(
+                        node_label_exprs
+                            .into_iter()
+                            .map(
+                                |LabelSelectorRequirement {
+                                     key,
+                                     operator,
+                                     values,
+                                 }| {
+                                    NodeSelectorRequirement {
+                                        key,
+                                        operator,
+                                        values,
+                                    }
+                                },
+                            )
+                            .collect(),
+                    ),
+                    ..NodeSelectorTerm::default()
+                }],
+            }),
+            ..NodeAffinity::default()
+        });
+
+        self.node_selector = node_labels.map(|node_labels| StackableNodeSelector {
+            node_selector: node_labels,
+        });
+        self.node_affinity = node_affinity;
+    }
+}
+
+#[derive(Clone, Debug, Eq, Deserialize, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct StackableNodeSelector {
+    #[serde(flatten)]
+    pub node_selector: BTreeMap<String, String>,
+}
+
+impl Atomic for StackableNodeSelector {}
+
+pub fn affinity_between_role_pods(
```

Comment:
This name makes more sense to me since it makes the purpose of this function clear. Also I would like not to hard-code the topology key in the framework.

```suggestion
pub fn default_role_affinity(
```

Reply:
Moved `kubernetes.io/hostname` into constant

Reply:
Regarding the name of the fn: I'm happy to change the name, but i think the "default" is a implementation details of the caller (the individual operators).
What do e.g. think of `role_pod_affinity` and `cluster_pod_affinity` below?